### ISSUE TITLE
Some games are extremly slow on OGL HW & SW

### ISSUE THREAD

**Issue by LeCharlatan** [PCSX2/pcsx2#2269]
***PCSX2 version :***
Stable 1.4.0
Dev 1.5.0 (latest)

***PCSX2 options :***
Using normal options. I tried default options, changing options, nothing solved the issue

***Plugins used :***

GSdx OGL HW/SW

***Note :*** 
I provide GS Dump (OGL HW) for each games since oddly, the issue only occurs on OGL. I tested game on another computer that uses Windows, tried DX11 and didn't get the slow-down.

Need for Speed - Most Wanted - Black Edition (PAL)
[gsdump-NFS-Most-Wanted-Black-Edition-PAL.zip](https://github.com/PCSX2/pcsx2/files/1639161/gsdump-NFS-Most-Wanted-Black-Edition-PAL.zip)
007 - Everything or Nothing (PAL)
[gsdump-007-Everything-or-Nothing-PAL.zip](https://github.com/PCSX2/pcsx2/files/1639163/gsdump-007-Everything-or-Nothing-PAL.zip)

***Description of the issue :***

Using GSdx OGL HW, 007 - Everything or Nothing and Need for speed Most Wanted Black Edition (both PAL) are very slow in-game and pcsx2's console is spaming such lines :

Gif Path[1] - MTGS Wait! [r=0x7ff770]
Gif Path[1] - MTGS Wait! [r=0x69cb30
Gif Path[1] - MTGS Wait! [r=0x7ff8f0]
Gif Path[1] - MTGS Wait! [r=0x782b80]
Gif Path[1] - MTGS Wait! [r=0x7ffac0]
Gif Path[1] - MTGS Wait! [r=0x7f42c0]
Gif Path[1] - MTGS Wait! [r=0x7ff6c0]
Gif Path[1] - MTGS Wait! [r=0x7668c0]
Gif Path[1] - MTGS Wait! [r=0x7ff390]
Gif Path[1] - MTGS Wait! [r=0x7ee8b0]

***How to reproduce the issue :***

Just play those games on OGL HW

***Last known version to work:***

I don't know, tested only latest stable and dev versions

***PC specifications:***

**OS :** Antergos Linux
**Desktop :** KDE (Plasma)
**CPU :** i5-4690K 3.50Ghz (Haswell)
**GPU :** Nvidia GTX 1070 8Go Vram (384.111)
**RAM :** 16Go

**Comment by lightningterror** [comment]
That's because OGL is more accurate than DX11.

You can probably get the same kind of performance if you disable blending, depth emulation, set crc to full.

**Comment by FlatOutPS2** [comment]
Make sure GSdx blending unit accuracy is not set to Ultra or Full. If it is put it back to Basic or High and make sure the Internal Resolution is not too high(try 3x max). With these settings your specs should be able to run at least EoN with OGL at decent speed for most of the game.

**Comment by LeCharlatan** [comment]
@lightningterror @FlatOutPS2 ,Hi both of you and thanks to answer so fast !

I tried to change options like you suggested but it's still slow and console is spaming messages like that : 

Gif Path[1] - MTGS Wait! [r=0x7ffbe0]
Gif Path[1] - MTGS Wait! [r=0x0]
Gif Path[1] - MTGS Wait! [r=0x7ffbf0]
Gif Path[1] - MTGS Wait! [r=0x0]
Gif Path[1] - MTGS Wait! [r=0x7ff8e0]
Gif Path[1] - MTGS Wait! [r=0x0]

I do not doubt about you but, such messages aren't odd ?

**GSdx OpenGL Options** 

![screenshot_20180117_152348](https://user-images.githubusercontent.com/22820335/35047652-e6a87fee-fb9a-11e7-8d0e-0d611862c1bc.png)

![screenshot_20180117_152356](https://user-images.githubusercontent.com/22820335/35047657-e97082e4-fb9a-11e7-9059-806b668e0128.png)

Result is better with 24 FPS (previous options was around 10 FPS)

**Comment by MrCK1** [comment]
Make sure you aren't using GSdx-dbg.

**Comment by FlatOutPS2** [comment]
You should check your Emulation Settings. Are you using any non-default EE/IOP, VUs, speedhacks or GameFixes settings?

**Comment by LeCharlatan** [comment]
@MrCK1 Hi, I'm not using GSdx-dbg :D 

@FlatOutPS2 I never change EE/IOP and VUs settings, they're by default (I checked in case they were not but they are), I never use SpeeHacks except for multicore CPU sometimes but it's not the case here. About GameFixes settings, I don't use any of them :x

**Comment by FlatOutPS2** [comment]
Then please post your emulog.txt(in the PCSX2 logs folder) contents after playing one of these games.

**Comment by LeCharlatan** [comment]
@FlatOutPS2 Here it is : 
[emuLog.txt](https://github.com/PCSX2/pcsx2/files/1639367/emuLog.txt)

**Comment by FlatOutPS2** [comment]
Yeah, the issue really seems to come down to the "MTGS Wait" sync warnings.
Not sure if there's anything you can do. @gregory38 might be able to help.

**Comment by MrCK1** [comment]
MTGS warnings only appear on debug builds, but I don't know how it works on Linux.

**Comment by LeCharlatan** [comment]
@gregory38 Help ! :D 
@MrCK1 Yes, you're right since Windows doesn't display it. However it's slow like Linux.

**PS :** I swear I don't use GSdx-dbg or I'm unaware of it !

**Comment by FlatOutPS2** [comment]
> PS : I swear I don't use GSdx-dbg or I'm unaware of it !

You aren't. The log doesn't mention any dev or debug details, so it must be in the release log on Linux.

**Comment by LeCharlatan** [comment]
@FlatOutPS2  Okay, I was sure at 99% now I'm sure at 100%, that's a relief :D 
@MrCK1 The same issue appear on stable 1.4.0 but no warning on console. So seems to display on the (latest) dev versions

**Comment by turtleli** [comment]
If I'm reading the code correctly then those messages only appear in dev or debug builds.
pcsx2/Gif_Unit.h:
```
if (IsDevBuild) {
	DevCon.WriteLn(Color_Red,   "Gif Path[%d] - MTGS Wait! [r=0x%x]", idx+1, getReadAmount());
	Gif_MTGS_Wait(isMTVU());
	DevCon.WriteLn(Color_Green, "Gif Path[%d] - MTGS Wait! [r=0x%x]", idx+1, getReadAmount());
	return;
}
```

**Comment by FlatOutPS2** [comment]
Oh yeah, it is a dev-build. The Onepad and SPU2 plugins mention it. The PCSX2 version at the top and other plugins don't say dev though, which is a bit confusing. Might be something to improve in the future.

**Comment by LeCharlatan** [comment]
@FlatOutPS2 
I don't use GSdx-dbg but to make it clear, the version I use is the latest dev version, compiled on Antergos Linux. The issue also happens on stable 1.4.0 as I described on the post

**Comment by FlatOutPS2** [comment]
No, you're not using a debug build, but it might be a release with debug info build. Which would explain why it isn't to be a debug build, but why it would still print the message.

**Comment by gregory38** [comment]
There are 3 linux build.
Debug : very slow
Dev: mean speed (IIRC default build)
Release:full speed

You seem to have a dev build. How did you build/install PCSX2 ?

**Comment by LeCharlatan** [comment]
Hi @gregory38 

Okay maybe I did something wrong so I'll explain how I Build it.

I download the latest sources here on github, then I build (compile) it throught the script "build.sh" that is inside sources with following command : 

./build.sh --no-simd --clean

That's all

**Comment by gregory38** [comment]
Yes you should use release flag. And why do you disable SIMD (sse/avx) optimization ? You should do it if you plan to run the build on a different machine.

**Comment by gregory38** [comment]
Then you should check that
1/ nvidia driver is installed
2/ that you enable multithread optimization when you launch PCSX2 (i.e. you use the small script to launch PCSX2 and not the binary directly)

**Comment by LeCharlatan** [comment]
Hi @gregory38 
Nvidia driver is installed. Using proprietary 384.111
I always launch PCSX2 by using the small script named "PCSX2-linux.sh"

I used the flag --no-simd because if I don't, it builds GSdx AVX2/AVX2 only not SSE2, SSE4.1...

I will build PCSX2 again with --release flag and will not use the --no-simd flag and report.
All that doesn't change the fact that those 2 EA Games (maybe more) are oddly slow using OpenGL even on Windows. When I tried on Windows, I used build from Orphis buildbot

**Comment by LeCharlatan** [comment]
Okay I report the test built using ./build.sh --release --clean

There are no warnings about "MTGS Wait" but it's still the same, very slow. I guess it only supress warnings

So finally it didn't change anything about the issue

**Comment by gregory38** [comment]
How do you launch PCSX2 ? Vsync ? Gsdx debug option ?

**Comment by LeCharlatan** [comment]
for the Vsync, do you mean the line "Wait for Vsync on Refresh" ? If so this is Disabled
I don't use GSdx debug options tab

**Comment by LeCharlatan** [comment]
@gregory38 
Well it was the mess with all my builds so maybe I did something wrong finally cause with the release build I can get better FPS using low settings in GSdx options !
NFS isn't playable tho, but 007 looks fine

**Comment by FlatOutPS2** [comment]
> NFS is not playable tho (30 to 45 FPS), 

You should be able to use skipdraw to (1 to 3) to increase the FPS in NFS. That game always had someframerate issues, just not as bad as you had before.
